front/app/components/IdeaCards/shared/Filters/TopicFilterDropdown.tsx

```diff
@@ -34,7 +34,17 @@ const TopicFilterDropdown = ({
   const { data: topics } = useTopics();
 
   const filteredTopics = useMemo(() => {
-    return topics?.data.filter((topic) => topicIds.includes(topic.id));
+    if (!topics) return [];
+
+    const topicsById = topics.data.reduce((acc, topic) => {
+      if (topicIds.includes(topic.id)) {
+        acc[topic.id] = topic;
+      }
+      return acc;
+    }, {});
+
+    // Return filtered topics in the order of topicIds
```

Comment:
Confused about this one, I thought we wanted to order by the back office order. Does the back office not allow you to order them and therefore doesn't it have an ordering field?

Reply:
Ah OK, get it!